### ISSUE TITLE
MenuItem sizing issue when using large modifier with nested items

### ISSUE THREAD

**Issue by chris-nolan** [palantir/blueprint#4893]
**Environment**

- Package version(s): 4.0.0-alpha.0
- Operating System: Windows
- Browser name and version: Edge Version 92.0.902.78 (Official build) (64-bit)

**Steps to reproduce**

- Hover over a MenuItem with nested MenuItem(s)

**Expected Result**

- Sub-menu opens with no sizing change to parent item

**Actual Result**

- Parent MenuItem size changes (looks like top/bottom padding reduces), which is particularly an issue when hovering at the lower edge of the item, and doesn't look great

**CodeSandbox Example**

https://codesandbox.io/s/blueprint-4-menu-issue-wmb12?file=/src/index.tsx

**Comment by adidahiya** [comment]
Good catch. This is an annoying styling bug. Here's a gif demonstrating the issue:

![2021-09-13 12 31 20](https://user-images.githubusercontent.com/723999/133122193-d6438cf7-82f6-4756-9be8-43f1f6dc50dd.gif)
